github/dependency_graph_snapshots.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+type DependencyRelationship string
+
+const (
+	DIRECT   DependencyRelationship = "direct"
+	INDIRECT                        = "indirect"
+)
+
+type DependencyScope string
+
+const (
+	RUNTIME     DependencyScope = "runtime"
+	DEVELOPMENT                 = "development"
+)
+
+type SnapshotCreationResult string
+
+const (
+	SUCCESS  SnapshotCreationResult = "SUCCESS"
+	ACCEPTED                        = "ACCEPTED"
+	INVALID                         = "INVALID"
+)
```

Comment:
Plese revert these changes.  In this repo, everything lives at the top-level namespace of the "github" module.  As a result, generic names like this (e.g. `SUCCESS`) are simply too general and therefore confusing to the user of this repo, since they are specifically only used for the DependencyGraph service.

Instead, we document using GoDoc-style comments what the values of certain fields can be (according to the official GitHub v3 API documentation).

Additionally, all your structs have names that are too general and need to be more specific.
Please take a browse through the code base of this repo and see how structs relating to specific endpoints and service have been names, and follow likewise.

I'm going to stop my review at this point so that you can please make the appropriate modifications.  Thanks.

Reply:
Alright, thanks for the feedback, will address the comments

Reply:
Hey @gmlewis, just addressed the comments. Thanks for the feedback!
Hopefully the comments around the possible values for the string structs are what you're expecting